include/json/forwards.h

```diff
@@ -13,11 +13,17 @@
 namespace Json {
 
 // writer.h
+class StreamWriter;
```

Comment:
I don't like the order in the writer.h and reader.h files, it doesn't really make sense to me.

However, since these are ordered the same as the files are I am happy with this patch.